prov/efa/src/rdm/efa_rdm_pke_utils.h

```diff
@@ -94,6 +94,71 @@ efa_rdm_pke_copy_from_hmem_iov(struct efa_mr *iov_mr, struct efa_rdm_pke *pke,
 	return copied;
 }
 
+/** 
+ * @brief This function either posts RDMA read, or sends a NACK packet when p2p
+ * is not available or memory registration limit was reached on the receiver.
+ *
+ * @param[in]    ep         endpoint
+ * @param[in]    pkt_entry  packet entry
+ * @param[in]    rxe        RX entry
+ *
+ * @return 0 on success, or a negative error code.
+ */
+static inline int
+efa_rdm_pke_post_remote_read_or_nack(struct efa_rdm_ep *ep,
+				     struct efa_rdm_pke *pkt_entry,
+				     struct efa_rdm_ope *rxe)
+{
+	int err = 0;
+	int pkt_type;
+
+	pkt_type = efa_rdm_pke_get_base_hdr(pkt_entry)->type;
```

Comment:
It seems we should have the same code for non-p2p and FI_ENOMR. We can refactor it to something like

```
err = 0
p2p_avail = efa_rdm_ep_use_p2p(ep, rxe->desc[0])
if (p2p_avail)
  err = efa_rdm_ope_post_remote_read_or_queue(rxe);
if (err == -FI_ENOMR || !p2p_avail) {
 // same fallback code.
}

```

Reply:
it seems non-p2p needs to return ENOTSUPP when the peer doesn't support nack protocol, and ENOMR case will return EAGAIN. So the refactor I proposed earlier may need some modification. But I still think we should have uniform code to handle the nack protocol path for both non-p2p and ENOMR